### ISSUE TITLE
Fieldname conversion when inserting a document should trigger an entry in the metadata

### ISSUE THREAD

**Issue by staunsbjerg** [Findwise/Hydra#269]
When inserting documents with period in the fieldname an entry in the metadata should be added in order to convert back to the original fieldname.

_example:_
input

``` json
{
  "myObj.field" : "somevalue"
}
```

output

``` json
{
  "contents" : {
    "myObj-field" : "somevalue"
  },
  "metadata" : {
    "renamedFields" : {
      "myObj-field" : "myObj.field",
      #... and many more
    }
  }
}
```

**Comment by remen** [comment]
Ahh... and then I would propose hiding this mongo-specific behavior from the stage, i.e., a (Local?)Document will use the original field names.

**Comment by laserval** [comment]
I agree with @remen, the conversion of field names should be part of `MongoDocument`.

LocalDocument should accept any field name (currently probably limited to valid JSON field names).

**Comment by staunsbjerg** [comment]
agreed that would be even better
